### PR TITLE
Feat/06 pdf a6 generator

### LINKED ISSUE #19: [Phase 6] PDF-Renderer (A6x4)
## Kontext
Mehrere Rechnungen sollen als A6-Blöcke auf einem A4-Blatt ausgegeben werden.

## Ziel / Scope
- Layout: 4 Rechnungen im A6-Format auf A4
- Schneidemarkierungen hinzufügen

## Akzeptanzkriterien / DoD
- [ ] PDF mit 4 Rechnungen erzeugt
- [ ] Layout korrekt
- [ ] Tests grün

## Tasks
- [ ] PDF Renderer erweitern
- [ ] Layout-Funktion schreiben
- [ ] Test PDF generieren
- [ ] README dokumentieren

## Risiken / Out-of-Scope
- Noch keine Druckintegration

### PR DESCRIPTION
**Pull request by wontknow**
## Zusammenfassung
A6 PDF Generator

## Verwandte Issues
Closes #19 

## Änderungen
---> Copilot Zusammenfassung
This pull request introduces a comprehensive set of updates focused on enhancing PDF generation capabilities and developer tooling for the Billino backend. The main highlights are the addition of a new API endpoint for generating A6-format invoice PDFs, and a suite of well-documented scripts for debugging, demonstration, and comparison of PDF layouts and generators. These changes improve both product features and developer experience.

**API Enhancements:**

* Added a new POST endpoint `/a6-invoices` in `pdfs.py` to generate and store multi-invoice PDFs in A6 format (laid out 2x2 on A4 pages). This includes validation, PDF generation using `PDFA6Generator`, and storage logic.
* Imported `PDFA6Generator` in `pdfs.py` to support the new A6 invoice PDF generation feature.

**Developer Tooling and Documentation:**

* Added a detailed `README.md` in the `scripts/` directory, documenting all available utility, demo, and comparison scripts, as well as describing the professional PDF design standards used.
* Introduced a module docstring for the `scripts` package for clarity and documentation purposes.

**PDF Generation Scripts:**

* Added demo and utility scripts:
  - [`demo_a6_generator.py`](diffhunk://#diff-517b036e0f72c07cf544aee0c229aafda4aa13bb07cf9d3c1caec1cd10908a17R1-R96): Generates a demo PDF with 4 realistic invoices in A6 layout.
  - [`debug_a6_layout.py`](diffhunk://#diff-3b36b6539ef4e7a3acab110e22a4cf69402c0b78e3d3abd4aaf152f7bbbcc65bR1-R84): Outputs margin and positioning calculations for A6 layouts, including an ASCII diagram.
  - [`compare_a6_generators.py`](diffhunk://#diff-0c05e55fe81fa8292601cad48d6a6331e95d1f539a0f6d1b6572a4251654b323R1-R115): Compares the original and simple A6 PDF generators, outputting file size and layout information.
  - [`demo_pdf_generation.py`](diffhunk://#diff-b22be4f60bd1bd7a63f21a2c9e7a2739665e76a9a84121beb3a535988d44e741R1-R134): Generates sample individual and summary invoice PDFs with professional design for testing and showcase purposes.

## Screenshots / Logs
(bei UI)

## Checkliste (DoD)
- [x] Funktionalität erfüllt (Akzeptanzkriterien)
- [x] Readme/Docs aktualisiert (falls nötig)
- [x] Keine Secrets/Keys committed
- [x] Lint/Build grün
- [x] Manuelle Tests (happy path & edge cases)
- [ ] CORS geprüft (Dev/Prod)  
- [x] PDF erzeugt & visuell geprüft  
- [ ] (optional) E-Rechnung Validierung dokumentiert

### CODE REVIEW

File: backend/services/pdf_a6_generator.py [hunk 1]
```diff
@@ -0,0 +1,479 @@
...
+                    self.styles["A6Address"],
+                ),
+            ]
+        ]
+
+        address_table = Table(
+            address_data, colWidths=[4 * cm, 4 * cm], rowHeights=[18 * mm]
+        )
+        address_table.setStyle(
+            TableStyle(
```
[the remark this answers is not in the document]
Comment (wontknow) [reply backend/services/pdf_a6_generator.py]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/wontknow/Billino/pull/40#discussion_r2483975592)
